components/tidb_query/src/rpn_expr/impl_time.rs

```diff
@@ -164,6 +163,22 @@ pub fn day_of_month(ctx: &mut EvalContext, t: &Option<DateTime>) -> Result<Optio
     Ok(Some(Int::from(t.day())))
 }
 
+#[rpn_fn(capture = [ctx])]
+#[inline]
+pub fn month_name(ctx: &mut EvalContext, t: &Option<DateTime>) -> Result<Option<Bytes>> {
+    match t {
+        Some(t) => {
+            if t.is_zero() && ctx.cfg.sql_mode.contains(SqlMode::NO_ZERO_DATE) {
```

Comment:
I think `ctx.cfg.sql_mode.contains(SqlMode::NO_ZERO_DATE)` should be removed, because no matter we are in `SqlMode::NO_ZERO_DATE` or not, we should report `NULL`

Reply:
I tried it in mysql.
If sql_mode = NO_ZERO_DATE is not set, there will be no warning in mysql.
```
mysql> show variables like "sql_mode";
+---------------+------------------------------------------------------------------------------------------+
| Variable_name | Value                                                                                    |
+---------------+------------------------------------------------------------------------------------------+
| sql_mode      | ONLY_FULL_GROUP_BY,STRICT_TRANS_TABLES,ERROR_FOR_DIVISION_BY_ZERO,NO_ENGINE_SUBSTITUTION |
+---------------+------------------------------------------------------------------------------------------+
1 row in set (0.01 sec)

mysql> select monthname('0000-00-00');
+-------------------------+
| monthname('0000-00-00') |
+-------------------------+
| NULL                    |
+-------------------------+
1 row in set (0.00 sec)
```